src/main/scala/io/iohk/ethereum/network/p2p/messages/Capability.scala

```diff
@@ -7,6 +7,22 @@ import io.iohk.ethereum.rlp.{RLPEncodeable, RLPException, RLPList, RLPSerializab
 case class Capability(name: String, version: Byte)
 
 object Capability {
+  def negotiate(c1: List[Capability], c2: List[Capability]): Option[Capability] = {
+    c1.intersect(c2).maxByOption(_.version) // FIXME ignores branches and other protocols
+  }
+
+  private val pattern = "(.*)/(\\d*)".r
+
+  def from(protocolVersion: String): Capability =
+    protocolVersion match {
+      case pattern(name, version) => Capability(name, version.toByte)
+      case _ => throw new RuntimeException(s"Unable to parse capability $protocolVersion")
```

Comment:
```suggestion
      case _ => throw new RuntimeException("Unable to parse capability {}", protocolVersion)
```

Reply:
I'm not convinced that RuntimeException accepts template strings.